### PR TITLE
[Weather command improvement]

### PR DESCRIPTION
**Pull request by robske110**
Almost finished!
(if you don't want the ability to set weather to all levels at once, merge this!)

### CODE REVIEW

**Comment by robske110** [comment]
This is NOT finished yet
(I have to test and translate)

**Comment by DaPigGuy** [comment]
Extra ] on line 49

File: src/pocketmine/command/defaults/WeatherCommand.php [hunk 1]
```diff
@@ -34,12 +33,28 @@ public function execute(CommandSender $sender, $currentAlias, array $args){
...
         }else{
             $seconds = 600*20;
+            $sender->sendMessage(new TranslationContainer(TextFormat::RED . "commands.weather.novalidnumber", $worldName));
         }
-
-        if($sender instanceof Player){
-            $level = $sender->getLevel();
-        }else{
-            $level = $sender->getServer()->getDefaultLevel();
+	    if(count($args) === 0 || count($args) === 1 || count($args) === 2)
```
Comment (robske110) [inline src/pocketmine/command/defaults/WeatherCommand.php]: Better way todo this?

Comment (inxomnyaa) [inline src/pocketmine/command/defaults/WeatherCommand.php]:
``` php
if(count($args) === 0 && !count($args) > 2)
```

or

``` php
if(count($args) === 0 && count($args) <= 2)
```

File: src/pocketmine/command/defaults/WeatherCommand.php [hunk 2]
```diff
@@ -34,12 +33,28 @@ public function execute(CommandSender $sender, $currentAlias, array $args){
             $seconds = (int) $args[1];
         }else{
             $seconds = 600*20;
+            $sender->sendMessage(new TranslationContainer(TextFormat::RED . "commands.weather.novalidnumber", $worldName));
```
Comment (inxomnyaa) [inline src/pocketmine/command/defaults/WeatherCommand.php]: commands.weather.invalidnumber

File: src/pocketmine/command/defaults/WeatherCommand.php (hunk 1 above)
Comment (inxomnyaa) [inline src/pocketmine/command/defaults/WeatherCommand.php]: edited

File: src/pocketmine/command/defaults/WeatherCommand.php [hunk 3]
```diff
@@ -24,9 +24,8 @@ public function execute(CommandSender $sender, $currentAlias, array $args){
             return true;
         }
 
-        if(count($args) > 2 || count($args) === 0){
+        if(count($args) > 3 || count($args) === 0){
```
Comment (thebigsmileXD) [inline src/pocketmine/command/defaults/WeatherCommand.php]: or >= 2

File: src/pocketmine/lang/locale/eng.ini [hunk 4]
```diff
@@ -325,6 +325,7 @@ pocketmine.plugin.eventError="Could not pass event '{%0}' to '{%1}': {%2} on {%3
 
 # -------------------- AdditionalLanguage --------------------
 commands.weather.usage=/weather [clear|rain|thunder] 
+commands.weather.worldnotfound=World '{%0}' could not be found!
 commands.weather.clear=Changing to clear weather
 commands.weather.rain=Changing weather to rainy weather
 commands.weather.thunder=Changing weather to thunder
```
Comment (inxomnyaa) [inline src/pocketmine/lang/locale/eng.ini:331]: commands.weather.invalidnumber=Invalid number given

Comment (robske110) [inline src/pocketmine/lang/locale/eng.ini:331]: It isn't invalid number it is "This is not an number" or something like that

Comment (inxomnyaa) [inline src/pocketmine/lang/locale/eng.ini:331]: then invalid argument

**Comment by robske110** [comment]
I will test this now
And still one question:Should i add, that if it is console command sender ALL levels should be cleared?

**Comment by inxomnyaa** [comment]
> And still one question:Should i add, that if it is console command sender ALL levels should be cleared?

Yes
or the one specified

**Comment by robske110** [comment]
Will do final tests today

**Comment by robske110** [comment]
Is there anything to do for me here?

**Comment by aodzip** [comment]
Is this finished?

**Comment by robske110** [comment]
If you don't want more features, yes
If you want more tell me
This is enough tested to work

**Comment by inxomnyaa** [comment]
The code style is messy. But it works

**Pull request merged**